app/src/main/java/ml/docilealligator/infinityforreddit/markdown/SpoilerParserPlugin.java

```diff
@@ -79,73 +79,69 @@ public void afterSetText(@NonNull TextView textView) {
-                int spanEnd = markdownStringBuilder.getSpanEnd(codeSpan);
-                if (spoilerStart + 2 <= spanBeginning && spanEnd <= spoilerEnd + 2) {
-                    markdownStringBuilder.delete(spoilerEnd, spoilerEnd + 2);
-                    markdownStringBuilder.delete(spoilerStart, spoilerStart + 2);
-                    SpoilerSpan spoilerSpan = new SpoilerSpan(textColor, backgroundColor);
-                    markdownStringBuilder.setSpan(spoilerSpan, spoilerStart, spoilerEnd - 2, Spanned.SPAN_EXCLUSIVE_EXCLUSIVE);
-                    offset += 4;
-                } else {
-                    break;
+        // Process all the found spoilers. We always want to delete the brackets
+        // because they are in matching pairs. But we want to apply SpoilerSpan
+        // only to the outermost spoilers because nested spans break revealing-hiding
+        int openingPosition = -1;
+        ArrayList<SpoilerBracket> brackets = new ArrayList<>();
+        for (SpoilerRange range : spoilers) {
+            brackets.add(new SpoilerBracket(range.start, true, range.nested));
+            brackets.add(new SpoilerBracket(range.end, false, range.nested));
+        }
+        //noinspection ComparatorCombinators as it requires api 24+
+        Collections.sort(brackets, (lhs, rhs) -> Integer.compare(lhs.position, rhs.position));
+
+        int offset = 0;
+        for (SpoilerBracket bracket: brackets) {
+            if (bracket.opening) {
+                int spoilerStart = bracket.position - offset;
+                if (!bracket.nested) {
+                    openingPosition = spoilerStart;
                 }
-            }
-
-            for (CodeBlockSpan codeBlockSpan : codeBlockSpans) {
-                int spanBeginning = markdownStringBuilder.getSpanStart(codeBlockSpan);
-                int spanEnd = markdownStringBuilder.getSpanEnd(codeBlockSpan);
-                if (spoilerStart + 2 <= spanBeginning && spanEnd <= spoilerEnd + 2) {
-                    markdownStringBuilder.delete(spoilerEnd, spoilerEnd + 2);
-                    markdownStringBuilder.delete(spoilerStart, spoilerStart + 2);
+                markdownStringBuilder.delete(spoilerStart, spoilerStart + 2);
+            } else {
+                int spoilerEnd = bracket.position - offset;
+                markdownStringBuilder.delete(spoilerEnd, spoilerEnd + 2);
+                if (!bracket.nested) {
                     SpoilerSpan spoilerSpan = new SpoilerSpan(textColor, backgroundColor);
-                    markdownStringBuilder.setSpan(spoilerSpan, spoilerStart, spoilerEnd - 2, Spanned.SPAN_EXCLUSIVE_EXCLUSIVE);
-                    offset += 4;
-                } else {
-                    break;
+                    markdownStringBuilder.setSpan(spoilerSpan, openingPosition, spoilerEnd,
+                            Spanned.SPAN_EXCLUSIVE_EXCLUSIVE);
                 }
             }
```

Comment:
This logic was incorrect and could call `delete` multiple times for a single spoiler. Caused crash for
```
>! `one` `two` `three` !<
```